### DISCUSSION TITLE
Welcome to cakephp-fileupload Discussions!

### DISCUSSION THREAD

**Discussion by maymeow** [MayMeow/cakephp-fileupload#7] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by TechnoPHP** [comment]
I installed the plugin and that went into 'vendor' folder. I am using a form to get user's account information and his profile pic in UsersController.php file. Can someone help me to set up.

**Reply by maymeow** [reply]
You can check https://dev.to/maymeow/uploading-files-with-cake-php-and-fileupload-plugin-303d as well as https://github.com/MayMeow/cakephp-fileupload/blob/master/README.md

PS: cakephp plugins like any other package that is installed with composer goes to the vendor folder.

By default is used local storage and uploaded files goes to the `ROOT . DS . 'storage' . DS` -> `/path/to/application/folder/storage`

**Comment by TechnoPHP** [comment]
I am not using doker. I followed this  https://github.com/MayMeow/cakephp-fileupload/blob/master/README.md   link.

I use UsersController to upload file and my upload() is in UsersController.php file also templates/Users/upload.php form file is there.

When I hit this url http://localhost:8765/users/upload, I get this error

Class 'App\Controller\UploadForm' not found

**Reply by maymeow** [reply]
You have to create new Modelless form by following https://book.cakephp.org/4/en/core-libraries/form.html.  or you can remove  `$uploadForm = new UploadForm();` and `$this->set(compact('uploadForm'));` rows from you Upload function, and just create form in view. By default component looking for file from `uploaded_file` field.

In README are missing examples for form/template files. (i can add them later today)

**Reply by maymeow** [reply]
Check once again https://dev.to/maymeow/uploading-files-with-cake-php-and-fileupload-plugin-303d 

It is not problem you are not using docker. Just skip that docker part others are same.  Scroll down to **Creating upload view and controller** There you have example for each, UploadForm and template.

Here is complete source code from tutorial above https://github.com/MayMeow/fileupload-tutorial

**Reply by TechnoPHP** [reply]
Ok, I removed those two lines from the upload function. That seems worked. Now there is another error. "The target directory storage does not exists or is not writable". My "storage" folder is in application's root parallel to src and webroot folder. I am using windows OS and the storage folder is writable.

**Comment by TechnoPHP** [comment]
File :  vendor/maymeow/file-upload/src/Storage/LocalStorageManager.php 

Function : move
$fileObject->moveTo($this->configuration->getConfig('storagePath') . $fileObject->getClientFilename());

this line is giving an error. if I print the path it is "storage/agent-registration-2.jpg"
As I said above my storage folder is parallel to src and webroot folder,

**Reply by maymeow** [reply]
I'll check this. (opened new issue #14 )

**Reply by TechnoPHP** [reply]
I think you need to use absolute path here adding ROOT constant, ie
ROOT .DS. $this->configuration->getConfig('storagePath') . $fileObject->getClientFilename()  
but that would be patch kind of thing. Just a hint.

**Reply by maymeow** [reply]
Default storage path is configured here https://github.com/MayMeow/cakephp-fileupload/blob/2d139a09b5f95f25984f694e2d82bad65af6bc14/src/Controller/Component/UploadComponent.php#L26

And loaded into Manager from StorageConfigInterface here https://github.com/MayMeow/cakephp-fileupload/blob/2d139a09b5f95f25984f694e2d82bad65af6bc14/src/Controller/Component/UploadComponent.php#L57 and  https://github.com/MayMeow/cakephp-fileupload/blob/2d139a09b5f95f25984f694e2d82bad65af6bc14/src/Storage/LocalStorageManager.php#L34

**Reply by maymeow** [reply]
problem can get up when user loads UploadComponent with his own `storagePath` property configuration and not using absolute path.  (as i said i will know later today)

**Reply by maymeow** [reply]
Ok tired it o windows and i had no problem with file upload

https://user-images.githubusercontent.com/3164256/150528818-13085fe2-988c-4517-94b8-2fa11d58e399.mp4

I have storage folder in root folder (same folder where src is) 

![Code_oF1dzGu7eF](https://user-images.githubusercontent.com/3164256/150528909-5fec7298-5738-4fc4-954f-440eeadc863a.png)

I'm loading component as follows

```php
/**
     * @throws \Exception
     */
    public function initialize(): void
    {
        parent::initialize(); // TODO: Change the autogenerated stub

        $this->loadComponent('FileUpload.Upload', [
            'fieldName' => 'file',
            # 'storagePath' => 'fileupload.tutorial', // bucket name
            'storage_type' => 'local'
        ]);

        $this->loadComponent('FileUpload.Download', [
            # 'storagePath' => 'fileupload.tutorial', // bucket name
            'storage_type' => 'local'
        ]);
    }
```
 
If you still have problem send me error log.

**Comment by maymeow** [comment]
For any other requests please create related conversation or use issues if you want to provide information about bug or you have idea how to make this project better.
